doc/devel/contributing.rst

```diff
@@ -21,44 +21,62 @@ also welcome to post feature requests or pull requests.
 Retrieving and installing the latest version of the code
 ========================================================
 
+When working on the Matplotlib source, setting up a `virtual
+environment
+<http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_ or a
+`conda environment <http://conda.pydata.org/docs/using/envs.html>`_ is
+recommended.
+
+.. warning::
+
+   If you already have a version of matplotlib installed, you will need to
+   uninstall it.
 
 We use `Git <https://git-scm.com/>`_ for version control and
 `GitHub <https://github.com/>`_ for hosting our main repository.
 
-You can check out the latest sources with the command::
+You can check out the latest sources with the command (see
+:ref:`set-up-fork` for more details)::
 
     git clone git@github.com:matplotlib/matplotlib.git
 
+and navigate to the matplotlib directory.
 
-After obtaining a local copy of the matplotlib source code (:ref:`set-up-fork`),
-navigate to the matplotlib directory and run the following in the shell::
-    
-    python setup.py develop
+To make sure the tests run locally you must build against the correct version
+of freetype.  To configure the build system to fetch and build it either export
+the env ``MPLLOCALFREETYPE`` as::
 
-or::
-  
-   pip install -v -e .
+  export MPLLOCALFREETYPE=1
 
+or copy :file:`setup.cfg.template` to :file:`setup.cfg` and edit to contain ::
 
-This installs matplotlib for development (i.e., builds everything and places the
-symbolic links back to the source code).
+  [test]
+  local_freetype = True
 
-.. warning::
 
-   If you already have a version of matplotlib installed, you will need to
-   uninstall it.
+To install Matplotlib (and compile the c-extensions) run the following
+command from the top-level directory ::
 
+    pip install -v -e ./
 
-.. note::
-  
-    If you decide to do install with ``python setup.py develop`` or ``pip
-    install -v -e``, you will have to rerun::
-   
-      python setup.py build
+This installs Matplotlib in 'editable/develop mode', i.e.,
+builds everything and places symbolic links back to the source code
```

Comment:
I don't think the symbolic link statement is true anymore. It adds a file in which the path to the library is written, and the path is explored at runtime.
(as I don't understand anything about the way python packaging is done, you should take this claim with a grain of salt…)